src/integrationTest/java/org/opensearch/test/framework/matcher/BulkResponseContainExceptionsAtIndexMatcher.java

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+package org.opensearch.test.framework.matcher;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import org.opensearch.action.bulk.BulkItemResponse;
+import org.opensearch.action.bulk.BulkResponse;
+
+import static java.util.Objects.requireNonNull;
+
+class BulkResponseContainExceptionsAtIndexMatcher extends TypeSafeDiagnosingMatcher<BulkResponse> {
```

Comment:
These matchers are really great, but I don't think they belong in the security plugin - I think they would be better suited as part of the high-level rest client, what do you think about moving them to that project?

Reply:
Looking back at this comment, its major scope creep! Lets move forward with the great work you've got here and we can enhance the HLRC in another effort